### ISSUE TITLE
Pass value of the `woocommerce_hold_stock_minutes` setting to `reserve_stock_for_order`

### ISSUE THREAD

**Issue by mikejolley** [woothemes/woocommerce#30612]
[The `reserve_stock_for_order` function seen here](https://github.com/woocommerce/woocommerce/blob/b19500728b4b292562afb65eb3a0c0f50d5859de/src/Checkout/Helpers/ReserveStock.php#L66-L67) can be unpredictable if `$minutes` are not provided, because it uses the `woocommerce_hold_stock_minutes` setting. Someone using this function may incorrectly assume the default is `60` when in fact it's based on store settings. If this setting is set to 0, it would completely prevent `reserve_stock_for_order` from working, even if an extension has a legitimate case for holding stock, unless it passes in a `$minutes` value.

`woocommerce_hold_stock_minutes` is a niche setting that should only affect core checkout, so I think we should instead pass this value to the `reserve_stock_for_order` method when called. This can be added here: https://github.com/woocommerce/woocommerce/blob/e4fec5b914d523ca7312506bd07f86ccc5cb30d5/includes/wc-stock-functions.php#L354

The default for $minutes can then be set to 60.

**Comment by rodelgc** [comment]
Hi @mikejolley ,

Thank you for opening the issue! I am adding the `needs developer feedback` label to this issue so that the Core team could take a look.

**Comment by jonathansadowski** [comment]
Thanks @mikejolley,

I just wanted to confirm, your suggestion is to modify `wc_reserve_stock_for_order` to have a new parameter for minutes with a default of 60, `$minutes = 60`, and pass that value on to the ReserveStock helper function, correct?

If so, I agree, I'll mark this as a task. If I'm misunderstanding, please correct.

**Comment by mikejolley** [comment]
Yes. That way checkout can define it's $minutes value based on the `woocommerce_hold_stock_minutes` setting, and the read of that setting can be removed from `ReserveStock::reserve_stock_for_order`.

**Comment by pmcpinto** [comment]
It looks like this is still relevant. I've marked it as ready for engineering triage.

Current evidence: The exact API cleanup requested in this issue is not complete on trunk. `ReserveStock::reserve_stock_for_order()` still defaults `$minutes` to `0` and falls back to `get_option( 'woocommerce_hold_stock_minutes', 60 )` when no value is passed, so direct helper callers still inherit a store setting that the issue says should only affect core checkout. The public `wc_reserve_stock_for_order()` wrapper still accepts only an order and calls the helper without passing a minutes value. Store API checkout now calls the wrapper rather than manually passing a hardcoded ten-minute draft-order value, and historical Blocks PR #5546 partially addressed payment-processing behavior, but the issue's proposed separation of core checkout setting from helper default remains unresolved. Open PR metadata has no hit for this issue.
